// media-chrome/dist/cjs/media-text-display.js
var r=Object.defineProperty;var g=Object.getOwnPropertyDescriptor;var p=Object.getOwnPropertyNames;var u=Object.prototype.hasOwnProperty;var b=i=>r(i,"__esModule",{value:!0});var y=(i,t)=>{for(var e in t)r(i,e,{get:t[e],enumerable:!0})},l=(i,t,e,n)=>{if(t&&typeof t=="object"||typeof t=="function")for(let o of p(t))!u.call(i,o)&&(e||o!=="default")&&r(i,o,{get:()=>t[o],enumerable:!(n=g(t,o))||n.enumerable});return i};var x=(i=>(t,e)=>i&&i.get(t)||(e=l(b({}),t,1),i&&i.set(t,e),e))(typeof WeakMap!="undefined"?new WeakMap:0);var L={};y(L,{default:()=>I});var c=require("./constants.js"),m=require("./utils/defineCustomElement.js"),a=require("./utils/server-safe-globals.js");const h=a.Document.createElement("template");h.innerHTML=`
  <style>
    :host {
      display: inline-flex;
      justify-content: center;
      align-items: center;
      box-sizing: border-box;
      background: var(--media-control-background, rgba(20,20,30, 0.7));
  
      padding: 10px;

      font-size: 14px;
      line-height: 24px;
      font-family: Arial, sans-serif;
      text-align: center;
      color: #ffffff;
      pointer-events: auto;
    }

    #container {
      /* NOTE: We don't currently have more generic sizing vars */
      height: var(--media-text-content-height, auto);
    }
  </style>
  <span id="container">
  <slot></slot>
  </span>
`;class f extends a.Window.HTMLElement{static get observedAttributes(){return[c.MediaUIAttributes.MEDIA_CONTROLLER]}constructor(){super();this.attachShadow({mode:"open"}),this.shadowRoot.appendChild(h.content.cloneNode(!0)),this.container=this.shadowRoot.querySelector("#container")}attributeChangedCallback(t,e,n){var o,d;if(t===c.MediaUIAttributes.MEDIA_CONTROLLER){if(e){const s=a.Document.getElementById(e);(o=s==null?void 0:s.unassociateElement)==null||o.call(s,this)}if(n){const s=a.Document.getElementById(n);(d=s==null?void 0:s.associateElement)==null||d.call(s,this)}}}connectedCallback(){var e;const t=this.getAttribute(c.MediaUIAttributes.MEDIA_CONTROLLER);if(t){const n=a.Document.getElementById(t);(e=n==null?void 0:n.associateElement)==null||e.call(n,this)}}disconnectedCallback(){var e;if(this.getAttribute(c.MediaUIAttributes.MEDIA_CONTROLLER)){const n=a.Document.getElementById(mediaControllerId);(e=n==null?void 0:n.unassociateElement)==null||e.call(n,this)}}}(0,m.defineCustomElement)("media-text-display",f);var I=f;module.exports=x(L);
