// media-chrome/dist/cjs/media-loading-indicator.js
var l=Object.defineProperty;var I=Object.getOwnPropertyDescriptor;var p=Object.getOwnPropertyNames;var y=Object.prototype.hasOwnProperty;var L=n=>l(n,"__esModule",{value:!0});var b=(n,t)=>{for(var i in t)l(n,i,{get:t[i],enumerable:!0})},M=(n,t,i,e)=>{if(t&&typeof t=="object"||typeof t=="function")for(let s of p(t))!y.call(n,s)&&(i||s!=="default")&&l(n,s,{get:()=>t[s],enumerable:!(e=I(t,s))||e.enumerable});return n};var E=(n=>(t,i)=>n&&n.get(t)||(i=M(L({}),t,1),n&&n.set(t,i),i))(typeof WeakMap!="undefined"?new WeakMap:0);var _={};b(_,{default:()=>T});var o=require("./constants.js"),m=require("./labels/labels.js"),D=require("./utils/defineCustomElement.js"),d=require("./utils/server-safe-globals.js");const r=d.Document.createElement("template"),v=`
<svg aria-hidden="true" viewBox="0 0 100 100">
  <path d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50">
    <animateTransform
       attributeName="transform"
       attributeType="XML"
       type="rotate"
       dur="1s"
       from="0 50 50"
       to="360 50 50"
       repeatCount="indefinite" />
  </path>
</svg>
`;r.innerHTML=`
<style>
:host {
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
}

#status {
  color: rgba(0,0,0,0);
  width: 0px;
  height: 0px;
}

:host slot[name=loading] > *,
:host ::slotted([slot=loading]) {
  opacity: 1;
  transition: opacity 0.15s;
}

:host(:not([is-loading])) slot[name=loading] > *, 
:host(:not([is-loading])) ::slotted([slot=loading]) {
  opacity: 0;
}

:host(:not([is-loading])) #status {
  display: none;
}

svg, img, ::slotted(svg), ::slotted(img) {
  width: var(--media-loading-icon-width, 100px);
  height: var(--media-loading-icon-height);
  fill: var(--media-icon-color, #fff);
  vertical-align: middle;
}
</style>

<slot name="loading">${v}</slot>
<div id="status" role="status" aria-live="polite">${m.nouns.MEDIA_LOADING()}</div>
`;const O=500;class u extends d.Window.HTMLElement{static get observedAttributes(){return[o.MediaUIAttributes.MEDIA_CONTROLLER,o.MediaUIAttributes.MEDIA_PAUSED,o.MediaUIAttributes.MEDIA_LOADING,"loading-delay"]}constructor(){super();const t=this.attachShadow({mode:"open"}),i=r.content.cloneNode(!0);t.appendChild(i)}attributeChangedCallback(t,i,e){var s,c,g;if(t===o.MediaUIAttributes.MEDIA_LOADING||t===o.MediaUIAttributes.MEDIA_PAUSED){const a=this.getAttribute(o.MediaUIAttributes.MEDIA_PAUSED)!=null,A=this.getAttribute(o.MediaUIAttributes.MEDIA_LOADING)!=null,h=!a&&A;if(!h)this.loadingDelayHandle&&(clearTimeout(this.loadingDelayHandle),this.loadingDelayHandle=void 0),this.removeAttribute("is-loading");else if(!this.loadingDelayHandle&&h){const f=+((s=this.getAttribute("loading-delay"))!=null?s:O);this.loadingDelayHandle=setTimeout(()=>{this.setAttribute("is-loading",""),this.loadingDelayHandle=void 0},f)}}else if(t===o.MediaUIAttributes.MEDIA_CONTROLLER){if(i){const a=d.Document.getElementById(i);(c=a==null?void 0:a.unassociateElement)==null||c.call(a,this)}if(e){const a=d.Document.getElementById(e);(g=a==null?void 0:a.associateElement)==null||g.call(a,this)}}}connectedCallback(){var i;const t=this.getAttribute(o.MediaUIAttributes.MEDIA_CONTROLLER);if(t){const e=d.Document.getElementById(t);(i=e==null?void 0:e.associateElement)==null||i.call(e,this)}}disconnectedCallback(){var i;if(this.loadingDelayHandle&&(clearTimeout(this.loadingDelayHandle),this.loadingDelayHandle=void 0),this.getAttribute(o.MediaUIAttributes.MEDIA_CONTROLLER)){const e=d.Document.getElementById(mediaControllerId);(i=e==null?void 0:e.unassociateElement)==null||i.call(e,this)}}}(0,D.defineCustomElement)("media-loading-indicator",u);var T=u;module.exports=E(_);
