// media-chrome/dist/cjs/media-chrome-range.js
var d=Object.defineProperty;var p=Object.getOwnPropertyDescriptor;var l=Object.getOwnPropertyNames;var k=Object.prototype.hasOwnProperty;var v=n=>d(n,"__esModule",{value:!0});var f=(n,t)=>{for(var e in t)d(n,e,{get:t[e],enumerable:!0})},y=(n,t,e,r)=>{if(t&&typeof t=="object"||typeof t=="function")for(let a of l(t))!k.call(n,a)&&(e||a!=="default")&&d(n,a,{get:()=>t[a],enumerable:!(r=p(t,a))||r.enumerable});return n};var w=(n=>(t,e)=>n&&n.get(t)||(e=y(v({}),t,1),n&&n.set(t,e),e))(typeof WeakMap!="undefined"?new WeakMap:0);var I={};f(I,{default:()=>x});var s=require("./constants.js"),m=require("./utils/defineCustomElement.js"),o=require("./utils/server-safe-globals.js");const g=o.Document.createElement("template"),u=`
  height: var(--thumb-height);
  width: var(--media-range-thumb-width, 10px);
  border: var(--media-range-thumb-border, none);
  border-radius: var(--media-range-thumb-border-radius, 10px);
  background: var(--media-range-thumb-background, #fff);
  box-shadow: var(--media-range-thumb-box-shadow, 1px 1px 1px transparent);
  cursor: pointer;
  transition: var(--media-range-thumb-transition, none);
  transform: var(--media-range-thumb-transform, none);
  opacity: var(--media-range-thumb-opacity, 1);
`,c=`
  width: var(--media-range-track-width, 100%);
  min-width: 40px;
  height: var(--track-height);
  border: var(--media-range-track-border, none);
  border-radius: var(--media-range-track-border-radius, 0);
  background: var(--media-range-track-background-internal, var(--media-range-track-background-color, #eee));

  box-shadow: var(--media-range-track-box-shadow, none);
  transition: var(--media-range-track-transition, none);
  cursor: pointer;
`;g.innerHTML=`
  <style>
    :host {
      --thumb-height: var(--media-range-thumb-height, 10px);
      --track-height: var(--media-range-track-height, 4px);

      position: relative;
      display: inline-block;
      vertical-align: middle;
      box-sizing: border-box;
      background: var(--media-control-background, rgba(20,20,30, 0.7));
      transition: background 0.15s linear;
      height: 44px;
      width: 100px;
      padding: 0 10px;

      pointer-events: auto;
    }

    /*
      Only show outline when keyboard focusing.
      https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo
    */
    :host-context([media-keyboard-control]):host(:focus),
    :host-context([media-keyboard-control]):host(:focus-within) {
      box-shadow: inset 0 0 0 2px rgba(27, 127, 204, 0.9);
    }

    :host(:hover) {
      background: var(--media-control-hover-background, rgba(50,50,60, 0.7));
    }

    input[type=range] {
      /* Reset */
      -webkit-appearance: none; /* Hides the slider so that custom slider can be made */
      background: transparent; /* Otherwise white in Chrome */

      /* Fill host with the range */
      min-height: 100%;
      width: var(--media-range-track-width, 100%); /* Specific width is required for Firefox. */

      box-sizing: border-box;
      padding: 0;
      margin: 0;
    }

    /* Special styling for WebKit/Blink */
    input[type=range]::-webkit-slider-thumb {
      -webkit-appearance: none;
      ${u}
      /* You need to specify a margin in Chrome, but in Firefox and IE it is automatic */
      margin-top: calc(calc(0px - var(--thumb-height) + var(--track-height)) / 2);
    }
    input[type=range]::-moz-range-thumb { ${u} }

    input[type=range]::-webkit-slider-runnable-track { ${c} }
    input[type=range]::-moz-range-track { ${c} }
    input[type=range]::-ms-track {
      /* Reset */
      width: 100%;
      cursor: pointer;
      /* Hides the slider so custom styles can be added */
      background: transparent;
      border-color: transparent;
      color: transparent;

      ${c}
    }

    /* Eventually want to move towards different styles for focus-visible
       https://github.com/WICG/focus-visible/blob/master/src/focus-visible.js
       Youtube appears to do this by de-focusing a button after a button click */
    input[type=range]:focus {
      outline: 0;
    }
    input[type=range]:focus::-webkit-slider-runnable-track {
      outline: 0;
    }

    input[type=range]:disabled::-webkit-slider-thumb {
      background-color: #777;
    }

    input[type=range]:disabled::-webkit-slider-runnable-track {
      background-color: #777;
    }

  </style>
  <input id="range" type="range" min="0" max="1000" step="1" value="0">
`;class b extends o.Window.HTMLElement{static get observedAttributes(){return[s.MediaUIAttributes.MEDIA_CONTROLLER]}constructor(){super();this.attachShadow({mode:"open"}),this.shadowRoot.appendChild(g.content.cloneNode(!0)),this.range=this.shadowRoot.querySelector("#range"),this.range.addEventListener("input",this.updateBar.bind(this))}attributeChangedCallback(t,e,r){var a,h;if(t===s.MediaUIAttributes.MEDIA_CONTROLLER){if(e){const i=o.Document.getElementById(e);(a=i==null?void 0:i.unassociateElement)==null||a.call(i,this)}if(r){const i=o.Document.getElementById(r);(h=i==null?void 0:i.associateElement)==null||h.call(i,this)}}}connectedCallback(){var e;const t=this.getAttribute(s.MediaUIAttributes.MEDIA_CONTROLLER);if(t){const r=o.Document.getElementById(t);(e=r==null?void 0:r.associateElement)==null||e.call(r,this)}this.updateBar()}disconnectedCallback(){var e;if(this.getAttribute(s.MediaUIAttributes.MEDIA_CONTROLLER)){const r=o.Document.getElementById(mediaControllerId);(e=r==null?void 0:r.unassociateElement)==null||e.call(r,this)}}updateBar(){const t=this.getBarColors();let e="linear-gradient(to right, ",r=0;t.forEach(a=>{a[1]<r||(e=e+`${a[0]} ${r}%, ${a[0]} ${a[1]}%,`,r=a[1])}),e=e.slice(0,e.length-1)+")",this.style.setProperty("--media-range-track-background-internal",e)}getBarColors(){const t=this.range,e=t.value/t.max*100;return[["var(--media-range-bar-color, #fff)",e],["var(--media-range-track-background-color, #333)",100]]}}(0,m.defineCustomElement)("media-chrome-range",b);var x=b;module.exports=w(I);
