// media-chrome/dist/cjs/media-settings-popup.js
var n=Object.defineProperty;var u=Object.getOwnPropertyDescriptor;var l=Object.getOwnPropertyNames;var h=Object.prototype.hasOwnProperty;var s=m=>n(m,"__esModule",{value:!0});var p=(m,e)=>{for(var o in e)n(m,o,{get:e[o],enumerable:!0})},b=(m,e,o,a)=>{if(e&&typeof e=="object"||typeof e=="function")for(let t of l(e))!h.call(m,t)&&(o||t!=="default")&&n(m,t,{get:()=>e[t],enumerable:!(a=u(e,t))||a.enumerable});return m};var w=(m=>(e,o)=>m&&m.get(e)||(o=b(s({}),e,1),m&&m.set(e,o),o))(typeof WeakMap!="undefined"?new WeakMap:0);var g={};p(g,{default:()=>x});var r=require("./utils/defineCustomElement.js"),i=require("./utils/server-safe-globals.js");const d=i.Document.createElement("template");d.innerHTML=`
  <style>
    :host {
      display: block;
      position: absolute;
      width: 300px;
      height: auto;
      right: 5px;
      bottom: 45px;
      padding: 10px;
      border: 1px solid #333;
      color: #ccc;
      background-color: #000;
    }
  </style>
  <slot></slot>

  <!-- This is too much for a menu... -->

  <media-chrome-panel>
  <media-chrome-menu>
    <media-chrome-submenu-menuitem label="Playback speed" value="1.2">
    </media-chrome-submenu-menuitem>
    <media-chrome-menuitem>Hello1</media-chrome-menuitem>
    <media-chrome-menuitem>Hello1</media-chrome-menuitem>
    <media-chrome-menuitem>Hello1</media-chrome-menuitem>
    <media-chrome-menuitem>Hello1</media-chrome-menuitem>
  </media-chrome-menu>
  <media-chrome-menu slot="menu">
    <media-chrome-menuitem>Normal</media-chrome-menuitem>
    <media-chrome-menuitem>1.5</media-chrome-menuitem>
  </media-chrome-menu>
`;class c extends i.Window.HTMLElement{constructor(){super();const e=this.attachShadow({mode:"open"});this.shadowRoot.appendChild(d.content.cloneNode(!0))}}(0,r.defineCustomElement)("media-settings-popup",c);var x=c;module.exports=w(g);
